apps/total-typescript/src/utils/trpc.ts

```diff
@@ -0,0 +1,16 @@
+import {createReactQueryHooks} from '@trpc/react'
+import type {inferProcedureOutput} from '@trpc/server'
+import {AppRouter} from 'server/routers/_app'
+import superjson from 'superjson'
+
+export const trpc = createReactQueryHooks<AppRouter>()
+
+export const transformer = superjson
+
+/**
+ * This is a helper method to infer the output of a query resolver
+ * @example type HelloOutput = inferQueryOutput<'hello'>
+ */
```

Comment:
all of the trpc stuff was [cut and paste from examples](https://github.com/trpc/examples-next-prisma-starter-websockets) with minor tweaks. notably I didn't bring over the websockets support from the example